File: app/src/main/java/com/openclassrooms/realestatemanager/RealEstateDao.java
package com.openclassrooms.realestatemanager;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface RealEstateDao {

    @Query("SELECT * FROM realestate ORDER BY id ASC")
    LiveData<List<RealEstate>> getAllRealEstate();

    @Query("SELECT * FROM realEstate WHERE id = :id")
    LiveData<RealEstate> getRealEstate(long id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void createRealEstate(RealEstate realEstate);

    @Update
    void updateRealEstate(RealEstate realEstate);
}

File: app/src/main/java/com/openclassrooms/realestatemanager/dao/RealEstateDao.kt
package com.openclassrooms.realestatemanager.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import com.openclassrooms.realestatemanager.model.RealEstate

@Dao
interface RealEstateDao {

    @Query("SELECT * FROM real_estate")
    fun getRealEstates(): LiveData<List<RealEstate>>

    @Query("SELECT * FROM real_estate WHERE id = :id")
    fun getRealEstate(id: Int): LiveData<RealEstate>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertRealEstate(realEstate: RealEstate)

    @Update
    suspend fun updateRealEstate(realEstate: RealEstate)
}